lib/individual/spend_time.dart

```diff
@@ -0,0 +1,489 @@
+                          filled: true,
+                          suffixIcon:
+                              Icon(Icons.arrow_drop_down), //icon of text field
+                          labelText:
+                              " Time", //labelStyle: TextStyle(fontSize:  MediaQuery.of(context).size.height*0.036)
+                          //label text of field
+                        ),
+
+                        readOnly:
+                            true, //set it true, so that user will not able to edit text
+                        onTap: () async {
+                          TimeOfDay? pickedTime = await showTimePicker(
+                            initialTime: TimeOfDay.now(),
+                            context: context,
+                          );
+
+                          if (pickedTime != null) {
+                            print(pickedTime.format(context)); //output 10:51 PM
+                            DateTime parsedTime = DateFormat.jm()
+                                .parse(pickedTime.format(context).toString());
+                            //converting to DateTime so that we can further format on different pattern.
+                            print(parsedTime); //output 1970-01-01 22:53:00.000
+                            String formattedTime =
+                                DateFormat('HH:mm:ss').format(parsedTime);
+                            print(formattedTime); //output 14:59:00
+                            //DateFormat() is from intl package, you can format the time on any pattern you need.
+
+                            setState(() {
+                              timeinput.text =
+                                  formattedTime; //set the value of text field.
+                            });
+                          } else {
+                            print("Time is not selected");
+                          }
+                        },
+                      ),
+                    ),
+                  ),
+                  SizedBox(height: MediaQuery.of(context).size.height * 0.01),
+                  Container(
+                    padding: EdgeInsets.all(0),
+                    decoration: BoxDecoration(
+                      color: Color(0xFF6CAFB4),
+                    ),
+                    child: ElevatedButton(
+                      style: ElevatedButton.styleFrom(
+                          shape: RoundedRectangleBorder(
+                              /*borderRadius: BorderRadius.circular(
+                                  MediaQuery.of(context).size.height * 0.02)*/
+                              ),
+                          primary: Color(0xFF6CAFB4)),
+                      child: Text(
+                        'Submit',
+                        style: TextStyle(
+                            fontWeight: FontWeight.w700, color: Colors.white),
+                      ),
+                      onPressed: (() {
+                        MessageDialog(
```

Comment:
This dialog has already been created.Use the widget from lib/shared/widget/message_dialog.dart and provide the necessary arguments. Will ensure that there is no code duplication.